pkg/kafka/policy.go

```diff
@@ -183,13 +233,12 @@ func (req *RequestMessage) ruleMatches(rule api.PortRuleKafka) bool {
 		return false
 	}
 
-	flowdebug.Log(log.WithFields(logrus.Fields{
-		fieldRequest: req.String(),
-		fieldRule:    rule,
-	}), "Matching Kafka rule")
+	if rule.APIKey != "" && rule.Role != "" {
+		return false
+	}
 
-	apiKey, isWildcard := rule.GetAPIKey()
-	if !isWildcard && apiKey != req.kind {
+	apiKey, isWildcard := rule.CheckAPIKeyRole(req.kind)
```

Comment:
Maybe have `CheckAPIKeyRole` just return `true` or `false` depending on whether it should be allowed instead of this complex logic?

Reply:
It just follows the same prototype of the earlier `rule.GetAPIKey` I can simplify it

Reply:
done